composer/core/time.py

```diff
@@ -0,0 +1,505 @@
+# Copyright 2021 MosaicML. All Rights Reserved.
+
+from __future__ import annotations
+
+import re
+import textwrap
+import warnings
+from typing import TYPE_CHECKING, Optional, Tuple, Union
+
+from composer.core.serializable import Serializable
+from composer.utils.string_enum import StringEnum
+
+if TYPE_CHECKING:
+    from composer.core.types import StateDict
+
+
+class TimeUnit(StringEnum):
```

Comment:
does `TimeUnit(str, Enum)` work here?

Reply:
The `StringEnum` class provides for additional functionality, such as being case-insensitive and for allowing comparisons between strings and the enum class. Would prefer to use it here as we use it elsewhere too.